cupy/cuda/compiler.py

```diff
@@ -111,6 +112,14 @@ def compile_with_cache(source, options=(), arch=None, cache_dir=None,
     if arch is None:
         arch = _get_arch()
 
+    cuda_path = os.getenv('CUDA_PATH', None)
+    if cuda_path is None:
+        warnings.warn('Please set the CUDA path ' +
```

Comment:
Yes, will add CUDA version checking and make that block effective when CUDA version >= 9.0.